### PR TITLE
Feat: signup page

### PR DESCRIPTION
**Pull request by Caleb00004**

### CODE REVIEW

File: pages/signup.tsx [hunk 1]
```diff
@@ -0,0 +1,420 @@
...
+    setSelected(newSection);
+  }
+
+  const creatorContent = (
+    <div className="text-[0.9em]">
+      <div className="flex flex-col mb-4">
+        <label htmlFor="fullName" className="mb-1 font-medium">
+          Full Name
+        </label>
+        <input
```
Comment (CeoFred) [inline pages/signup.tsx:44]: Please create an input component.. don't repeat yourself.
things that are reusable, make it a component.

File: pages/signup.tsx [hunk 2]
```diff
@@ -0,0 +1,420 @@
...
+      </div>
+      <div className="flex items-start gap-2 mb-4">
+        <input type="radio" className="mt-1" />
+        <label className="p-0 text-slate-600">
+          I consent to the collection and processing of my personal data in line
+          with data regulations as described in the{' '}
+          <span className="font-semibold text-[#5241AA]">Privacy Policy</span>
+        </label>
+      </div>
+      <button
```
Comment (CeoFred) [inline pages/signup.tsx:120]: I spoke about using literal buttons, please make this a component.

**Review by CeoFred — CHANGES_REQUESTED** [review]

File: components/ui/organisms/Button.tsx [hunk 3]
```diff
@@ -0,0 +1,11 @@
+export default function Button({ text, callBack }) {
```
Comment (CeoFred) [inline components/ui/organisms/Button.tsx:1]: Did you skip pre-commit hooks?? this component is not properly typed. onClick handler props should not be called callBack, that's vague. use same name as event

File: components/ui/organisms/SelectInput.tsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  options,
+}: SelectInputProps): JSX.Element {
+  const option = options.map((item) => (
+    <option value={item.value}>{item.label}</option>
+  ));
+
+  return (
+    <div className="flex flex-col mb-4">
+      <label className="mb-1 font-medium">{text}</label>
+      <select className="border py-2 px-2 rounded-md text-slate-600">
```
Comment (CeoFred) [inline components/ui/organisms/SelectInput.tsx:22]: No onChange handler.

File: components/ui/organisms/TextInput.tsx
Comment (CeoFred) [inline components/ui/organisms/TextInput.tsx:1]: Rename file to Input

File: components/ui/organisms/TextInput.tsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  id,
+}: InputProps) {
+  // console.log(setterFunc)
+  return (
+    <div className="flex flex-col mb-4">
+      <label htmlFor={id} className="mb-1 font-medium">
+        {text}
+      </label>
+      <input
+        required
```
Comment (CeoFred) [inline components/ui/organisms/TextInput.tsx:25]: this attribute cannot be always true, allow it to come from props and also have a default value.

File: components/ui/organisms/TextInput.tsx [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+        value={value}
+        id={id}
+        onChange={(e) =>
+          setterFunc((prev) => ({ ...prev, [id]: e.target.value }))
+        }
+        className="border py-2 px-2 rounded-md"
```
Comment (CeoFred) [inline components/ui/organisms/TextInput.tsx:31]: It's possible to override a className, we should be able to accept a className prop too.

File: pages/signup.tsx [hunk 7]
```diff
@@ -0,0 +1,387 @@
...
+        id="password"
+        type="password"
+        setterFunc={setFormData}
+        value={formData.password}
+      />
+
+      <p className="font-medium">Are you a software developer ?</p>
+      <div className="flex gap-4 py-3">
+        <div className="flex gap-1">
+          <input
```
Comment (CeoFred) [inline pages/signup.tsx:76]: You still have a native input element here. create a component. or find a way to use existing input component to render a radio input. you can have the input component export different input types

File: pages/signup.tsx [hunk 8]
```diff
@@ -0,0 +1,387 @@
...
+                    selected !== 'creator'
+                      ? {
+                          borderColor: 'white',
+                          border: 'solid',
+                          borderWidth: '1px',
+                        }
+                      : { border: 'none' }
+                  }
+                >
+                  <input
```
Comment (CeoFred) [inline pages/signup.tsx:228]: should be a component, this is going to be used somewhere else again.

Comment (CeoFred) [inline pages/signup.tsx:228]: should be a component, this is going to be used somewhere else again.

File: pages/signup.tsx [hunk 9]
```diff
@@ -0,0 +1,387 @@
...
+          </div>
+        </div>
+        <div className="flex h-[80vh] md:flex-1 md:h-screen bg-white py-5 pt-1 px-6 md:pt-0 md:px-10 lg:px-16 z-[2]">
+          <div className=" w-full mt-10 mb-7 md:mb-auto md:mt-[6.2em] ">
+            <div className="flex flex-col-reverse justify-start items-start md:flex-row md:justify-between">
+              <h1 className="text-[1.8em] text-[#6651D4] font-bold pb-4">
+                Create new account
+              </h1>
+
+              {!isCreator && nextBusiness && (
```
Comment (CeoFred) [inline pages/signup.tsx:370]: what the purpose of the variable `nextBusiness`?

**Review by CeoFred — CHANGES_REQUESTED** [review]

File: pages/signup.tsx (hunk 9 above)
Comment (Caleb00004) [reply pages/signup.tsx:370]: The business section has 2 forms, so the nextBusiness is to navigate to the next form and vice versa.

**Pull request closed without merging**